src/nvim/ui.coffee

```diff
@@ -73,15 +74,31 @@ class UI extends EventEmitter
     @cursor.style.height = @char_height + 'px'
 
     if config.blink_cursor
-      @cursor.classList.add('blink')
+      @startCursorBlink()
     else
-      @cursor.classList.remove('blink')
+      @stopCursorBlink()
+
+  startCursorBlink: ->
+    if config.blink_cursor
```

Comment:
The `if` guards is not consistent for all the functions (and their calls) related to cursors.
For example, it would be better to check `config.blink_cursor` before any call to these functions, but it should not be ever checked inside these functions.